x/posts/keeper/msg_server.go

```diff
@@ -498,3 +498,92 @@ func (m msgServer) UpdateParams(goCtx context.Context, msg *types.MsgUpdateParam
 
 	return &types.MsgUpdateParamsResponse{}, nil
 }
+
+// MovePost defines the rpc method for Msg/MovePost
+func (k msgServer) MovePost(goCtx context.Context, msg *types.MsgMovePost) (*types.MsgMovePostResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Make sure the post exists
+	post, found := k.GetPost(ctx, msg.SubspaceID, msg.PostID)
+	if !found {
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "post with id %d does not exist", msg.PostID)
+	}
+
+	// Check if the target subspace exists
+	if !k.HasSubspace(ctx, msg.TargetSubspaceID) {
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "subspace with id %d not found", msg.TargetSubspaceID)
+	}
+
+	// Check if the target section exists
+	if !k.HasSection(ctx, msg.TargetSubspaceID, msg.TargetSectionID) {
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "subspace section with id %d not found", msg.TargetSectionID)
+	}
+
+	// Check the sender is the owner of the post
+	if post.Author != msg.Owner {
+		return nil, errors.Wrapf(sdkerrors.ErrInvalidRequest, "you are not the owner of the post")
+	}
+
+	// Check the permission to move the post
+	if !k.HasPermission(ctx, msg.TargetSubspaceID, msg.TargetSectionID, msg.Owner, types.PermissionWrite) {
+		return nil, errors.Wrap(subspacestypes.ErrPermissionDenied, "you don't have write permission on the target section")
+	}
+
+	// Get the next post id
+	newPostID, err := k.GetNextPostID(ctx, msg.TargetSubspaceID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Update the post and validate it
+	updateTime := ctx.BlockTime()
+	postMove := types.NewPostMove(msg.TargetSubspaceID, msg.TargetSectionID, newPostID, updateTime)
+	updatedPost := postMove.Update(post)
+	err = k.ValidatePost(ctx, updatedPost)
+	if err != nil {
+		return nil, err
+	}
+
+	k.Keeper.SavePost(ctx, updatedPost)
+
+	// Update the id for the next post
+	k.SetNextPostID(ctx, msg.TargetSubspaceID, newPostID+1)
+
+	// Move all post attachments
+	newAttachmentID := uint32(0)
+	k.IteratePostAttachments(ctx, msg.SubspaceID, msg.PostID, func(attachment types.Attachment) (stop bool) {
+		newAttachmentID++
+		attachmentMove := types.NewAttachmentMove(msg.TargetSubspaceID, newPostID, newAttachmentID)
```

Comment:
To make this more clear, I think we should use the following notation instead:
```suggestion
		attachmentMove := types.NewAttachmentMove(updatedPost.SubspaceID, updatedPost.ID, newAttachmentID)
```